p2p/net/conn.go

```diff
@@ -343,40 +340,21 @@ var (
 )
 
 func (c *FormattedConnection) setupIncoming(ctx context.Context, timeout time.Duration) error {
-	be := make(chan struct {
-		b []byte
-		e error
-	})
-
-	go func() {
-		// TODO: some other way to make sure this groutine closes
-		err := c.deadliner.SetReadDeadline(time.Now().Add(timeout))
-		if err != nil {
-			be <- struct {
-				b []byte
-				e error
-			}{b: nil, e: err}
-			return
-		}
-		msg, err := c.r.Next()
-		be <- struct {
-			b []byte
-			e error
-		}{b: msg, e: err}
-		err = c.deadliner.SetReadDeadline(time.Time{}) // disable read deadline
-		if err != nil {
-			c.logger.With().Warning("could not set a read deadline", log.Err(err))
-		}
-	}()
-
-	msgbe := <-be
-	msg := msgbe.b
-	err := msgbe.e
-
+	err := c.deadliner.SetReadDeadline(time.Now().Add(timeout))
```

Comment:
this doesn't need to be in a goroutine. we are waiting for completion immediately after goroutine is spawned